jgrapht-core/src/main/java/org/jgrapht/alg/matching/DefaultMatching.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2016-2016, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.matching;
+
+import java.io.Serializable;
+import java.util.Set;
+
+import org.jgrapht.alg.interfaces.MatchingAlgorithm.Matching;
+
+/**
+ * A default implementation of the matching interface.
+ * 
+ * @param <E> the graph edge type
+ *
+ * @author Dimitrios Michail
+ */
+class DefaultMatching<E>
```

Comment:
I prefer to keep the default implementation in a separate class in order to keep it package-private. This way the user is forced to use the interface and not the Data-Transfer-Object. Moreover, this way we can use only one DTO for other interfaces which are also weighted edge sets. Similarly for vertex sets such as cliques, vertex-covers, etc.

Another argument for this is that it keeps the public footprint smaller and it is easier to keep compatibility between versions of the library.

I would vote to go the other way and move the MaximumFlowImpl in the flow packages as package-private. Of course, if both you and @jsichi prefer to use the flow pattern, we can do that.